library/tiqr/Tiqr/OcraService/Abstract.php

```diff
@@ -17,55 +17,39 @@
  * @copyright (C) 2010-2012 SURFnet BV
  */
 
-require_once("Tiqr/OcraService/Interface.php");
+use Psr\Log\LoggerInterface;
+
+require_once(__DIR__ . '/../OATH/OCRAParser.php');
 
-/**
- * The abstract class that defines what a ocra service class should implement.
- *
- * @author lineke
- *
- */
 abstract class Tiqr_OcraService_Abstract implements Tiqr_OcraService_Interface
 {
-    /**
-     * Verify the response
-     * Override in child class to implement this method if this is the verification method to use
-     *
-     * @param string $response
-     * @param string $userSecret
-     * @param string $challenge
-     * @param string $sessionKey
-     *
-     * @return boolean True if response matches, false otherwise
-     */
-    public function verifyResponseWithSecret($response, $userSecret, $challenge, $sessionKey)
-    {
-
-    }
+    /** @var OATH_OCRAParser */
+    protected $_ocraParser;
+    /** @var String */
+    protected $_ocraSuite;
+    /** @var LoggerInterface  */
+    protected $logger;
 
     /**
-     * Verify the response
-     * Override in child class to implement this method if this is the verification method to use
-     *
-     * @param string $response
-     * @param string $userId
-     * @param string $challenge
-     * @param string $sessionKey
-     *
-     * @return boolean True if response matches, false otherwise
+     * @throws Exception
      */
-    public function verifyResponseWithUserId($response, $userId, $challenge, $sessionKey)
-    {
+    public function __construct(array $config, LoggerInterface $logger) {
+        $this->logger = $logger;
```

Comment:
Is the logger used in this service? Or is it added to distribute it to its ancestors?

Reply:
Yes, the logger is used by both concrete classes